Bio/Seq.py

```diff
@@ -1280,6 +1281,54 @@ def ungap(self, gap=None):
             raise ValueError("Unexpected gap character, {0!r}".format(gap))
         return Seq(str(self).replace(gap, ""), alpha)
 
+    def join(self, other):
+        """Return a merge of the sequences in other, spaced by the sequence from self.
+
+        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+
+        >>> spacer = Seq('NNNNN')
+        >>> seqlist = [Seq("AAA"), Seq("TTT"), Seq("PPP")]
+        >>> concatenated = spacer.join(seqlist)
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable
+        are not Seq or String objects
+        """
+        if not isinstance(other, collections.Iterable):  # doesn't detect single strings
+            raise ValueError("Input must be an iterable")
+        if isinstance(other, basestring):
+            raise ValueError("Input must be an iterable")
+        from Bio.SeqRecord import SeqRecord  # Lazy to avoid circular imports
+        temp_data = ""
+        # if the spacer is empty it will initially default none
+        if self._data == "":
+            a = None
+        else:
+            a = self.alphabet
+        for c in other:
+            if isinstance(c, SeqRecord):
+                return NotImplemented
```

Comment:
I think you should only use ``return NotImplemented`` when implementing binary special methods.

Should this be ``raise NotImplementedError`` or better ``raise TypeError`` instead?

Reply:
In the original issue posted, widdowquinn used 
`seqlist = list(SeqIO.parse('myfile.fasta', 'fasta'))
concatenated = spacer.join(seqlist)` 
in his example usage of the method, so I thought it might be nice to keep a NotImplemented error( will be changed to raise NotImplementedError) instead of TypeError so there is an open door to use SeqRecords in the future.

Reply:
I thought we agreed in the discussion on #1731 that the ``Seq`` & ``SeqRecord`` mixing was unintentional. Using ``NotImplementedError`` suggests we do plan to implement this at some point, and right now I don't think we should. So a ``TypeError`` seems best here.

Reply:
Ok, my error. I changed it to a TypeError.